packages/dataviews/src/types.ts

```diff
@@ -80,6 +80,16 @@ export type Field< Item > = {
 	 */
 	render?: ComponentType< { item: Item } >;
 
+	/**
+	 * Callback used to sort the field.
+	 */
+	sort?: (
+		a: Item,
+		b: Item,
+		direction: SortDirection,
+		field?: any
```

Comment:
This is where I'm having issues: as soon as I add a type (Field/NormalizedField) here, the TypeScript parser complains.

Reply:
There's something weird here, to define the "sort" function, you need the "field" object but the field object contains the "sort" function. Why did you need "field" as an argument here?

Reply:
It is for the sort function defined by the type (see [integer.tsx](https://github.com/WordPress/gutenberg/pull/64064/files#diff-ea641593eb6d360ee4bcf491ea3678b942e42b9354e0fe7e81bae3d08902e9a8R12)) to be able to convert the item into the value:

```js
const valueA = field.getValue( { item: itemA } );
```

Reply:
I think the sort function defined by the type and the one defined by the field can be slightly different. For the field there's no need to pass the `field` but yeah it should be fine to keep the `field` for the field type one, or just pass `getValue` directly.

Reply:
Passing `getValue` makes sense https://github.com/WordPress/gutenberg/pull/64064/commits/bcf814aaa9ccf000db6d025e3175cfde27990ae5 Thanks!